kolibri/plugins/learn/assets/src/views/card-grid/grid-item/index.vue

```diff
@@ -1,23 +1,31 @@
 <template>
 
-  <router-link :to="link" class="link-block">
-    <div class="wrapper">
-      <div class="card-thumbnail">
-        <slot/>
-        <div class="progress-icon-wrapper">
-          <progress-icon :progress="progress"/>
-        </div>
+  <router-link :to="link" class="card">
+
+    <div class="card-thumbnail" :style="backgroundImg">
+      <content-icon v-if="!thumbnail" :kind="kind" class="card-thumbnail-backup"/>
+      <div class="card-progress-icon-wrapper">
+        <progress-icon :progress="progress"/>
+      </div>
+
+      <div class="card-content-icon-background" :class="backgroundClass"></div>
+      <div class="card-content-icon-wrapper">
+        <content-icon :kind="kind" class="card-content-icon"/>
       </div>
-      <div class="card-content">
-        <div class="text">
-          <content-icon
-          class="content-icon"
-          v-if="kind"
-          :kind="kind"/>
-          {{ title }}
+
+      <div class="card-progress-bar-wrapper">
+        <div
+          class="card-progress-bar"
+          :style="{ width: `${progress * 100}%` }"
+          :class="{ 'card-progress-bar-mastered': mastered, 'card-progress-bar-progress': inProgress }">
```

Comment:
Better to handle this style logic as an object inside of computed props? So that it ends up looking like:
`:class=progressBarStyles` with the object being generated in the computed property